Sources/TuistKit/Services/InitService.swift

```diff
@@ -92,23 +98,42 @@ class InitService {
 
         let directories = try templatesDirectoryLocator.templateDirectories(at: path)
         if let templateName = templateName {
-            guard
-                let templateDirectory = directories.first(where: { $0.basename == templateName })
-            else { throw InitServiceError.templateNotFound(templateName) }
-            let template = try templateLoader.loadTemplate(at: templateDirectory)
-            let parsedAttributes = try parseAttributes(
-                name: name,
-                platform: platform,
-                requiredTemplateOptions: requiredTemplateOptions,
-                optionalTemplateOptions: optionalTemplateOptions,
-                template: template
-            )
+            var template: Template?
+            if templateName.isURL {
+                try fileHandler.inTemporaryDirectory { temporaryDirectory in
+                    let templateDirectoryName = temporaryDirectory.appending(.init("\(templateName)")).url.lastPathComponent
+                    let templateDirectory = temporaryDirectory.appending(.init("\(templateDirectoryName)"))
+                    try fileHandler.createFolder(templateDirectory)
+                    try gitHandler.clone(url: templateName, to: templateDirectory)
+                    template = try templateLoader.loadTemplate(at: templateDirectory)
+                    guard let template = template else { return }
+                    try generateTemplate(
```

Comment:
I'd move this under the current `if else` along with [this](https://github.com/tuist/tuist/pull/3796/files#diff-9dd0b8d32d62ac9dbc9630a275b599bab8e9f93cd74777ea102bd18dda17e3e5R127) as those calls are currently redundant.